algorithms/src/sorting/impl/Kokkos_SortByKeyImpl.hpp

```diff
@@ -87,7 +87,18 @@ static_assert_is_admissible_to_kokkos_sort_by_key(const ViewType& /* view */) {
                 "LayoutRight, LayoutLeft or LayoutStride.");
 }
 
+// For the fallback implementation for sort_by_key using Kokkos::sort, we need
+// to consider if Kokkos::sort defers to the fallback implementation that copies
+// the array to the host and uses std::sort. This decision is made in
+// impl/Kokkos_SortImpl.hpp and this type trait decides at runtime when it is
```

Comment:
> this type trait decides at runtime when it is safe to assume that Kokkos::sort doesn't manually copy to the host

I don't understand this sentence.  There is no more trait and the variable template does not "decide" anything.  Do you mean it is used somewhere to determine what to do?

Reply:
Please improve/update the comment

Reply:
d5c747872f1f2d77b765462c4230408e4481720d